Src/Support/GoogleApis/Apis/[Media]/Upload/ResumableUpload.cs

```diff
@@ -390,22 +432,101 @@ public Task<IUploadProgress> UploadAsync()
+        /// </remarks>
+        /// <param name="cancellationToken">A cancellation token to cancel the asynchronous operation.</param>
+        public Task<IUploadProgress> ResumeAsync(CancellationToken cancellationToken)
+        {
+            return ResumeAsync(cancellationToken, null);
+        }
+        /// <summary>
+        /// Asynchronously resumes the upload from the last point it was interrupted. 
+        /// Use when resuming and the program was restarted.
+        /// Implemented only for ContentStreams where .CanSeek is True.
+        /// </summary>
+        /// <remarks>
+        /// In your application's UploadSessionData Event Handler, store UploadUri.AbsoluteUri property value (resumable session URI string value)
+        /// to persistent storage for use with Resume() or ResumeAsync() upon a program restart.
+        /// It is strongly recommended that the FullPathFilename of the media file that is being uploaded is saved also so that a subsequent execution of the
+        /// program can compare the saved FullPathFilename value to the FullPathFilename of the media file that it has opened for uploading.
+        /// You do not need to seek to restart point in the ContentStream file.
+        /// </remarks>
+        /// <param name="uploadUri">VideosResource.InsertMediaUpload UploadUri property value that was saved to persistent storage during a prior execution.</param>
+        public Task<IUploadProgress> ResumeAsync(Uri uploadUri)
         {
+            return ResumeAsync(CancellationToken.None, uploadUri);
+        }
+        /// <summary>
+        /// Asynchronously resumes the upload from the last point it was interrupted. 
+        /// Use when the program was restarted and you wish to resume the upload that was in progress when the program was halted.
+        /// Implemented only for ContentStreams where .CanSeek is True.
+        /// </summary>
+        /// <remarks>
+        /// In your application's UploadSessionData Event Handler, store UploadUri.AbsoluteUri property value (resumable session URI string value)
+        /// to persistent storage for use with Resume() or ResumeAsync() upon a program restart.
+        /// It is strongly recommended that the FullPathFilename of the media file that is being uploaded is saved also so that a subsequent execution of the
+        /// program can compare the saved FullPathFilename value to the FullPathFilename of the media file that it has opened for uploading.
+        /// You do not need to seek to restart point in the ContentStream file.
+        /// </remarks>
+        /// <param name="cancellationToken">A cancellation token to cancel the asynchronous operation.</param>
+        /// <param name="uploadUri">VideosResource.InsertMediaUpload UploadUri property value that was saved to persistent storage during a prior execution.</param>
+        public async Task<IUploadProgress> ResumeAsync(CancellationToken cancellationToken, Uri uploadUri)

```

Comment:
Can we reorder these parameters? It's weird having the cancellation token *before* other parameters.